src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs

```diff
@@ -425,69 +452,132 @@ public static CultureInfo CurrentCulture
                     throw new ArgumentNullException(nameof(value));
                 }
 
-#if ENABLE_WINRT
-                WinRTInteropCallbacks callbacks = WinRTInterop.UnsafeCallbacks;
-                if (callbacks != null && callbacks.IsAppxModel())
+#if FEATURE_APPX
+                if (ApplicationModel.IsUap)
                 {
-                    callbacks.SetGlobalDefaultCulture(value);
-                    return;
+                    if (SetCultureInfoForUserPreferredLanguageInAppX(value))
+                    {
+                        // successfully set the culture, otherwise fallback to legacy path
+                        return; 
+                    }
                 }
 #endif
```

Comment:
This obviously differs in approach in CoreRT and CoreCLR. I assume the easiest way to resolve this is to split it up, but maybe there's a way to share the code.